control-plane/pkg/reconciler/broker/namespaced_controller.go

```diff
@@ -115,7 +116,18 @@ func NewNamespacedController(ctx context.Context, watcher configmap.Watcher, env
 
 	reconciler.Resolver = resolver.NewURIResolverFromTracker(ctx, impl.Tracker)
 	reconciler.IPsLister = prober.NewIPListerWithMapping()
-	reconciler.Prober = prober.NewAsync(ctx, http.DefaultClient, env.IngressPodPort, reconciler.IPsLister.List, impl.EnqueueKey)
+
+	features := feature.FromContext(ctx)
+	caCerts, err := reconciler.getCaCerts()
+	if err != nil && (features.IsStrictTransportEncryption() || features.IsPermissiveTransportEncryption()) {
+		// We only need to warn here as the broker won't reconcile properly without the proper certs because the prober won't succeed
+		logger.Warn("Failed to get CA certs when at least one address uses TLS", zap.Error(err))
+	}
+
+	reconciler.Prober, err = prober.NewComposite(ctx, env.IngressPodPort, env.IngressPodTlsPort, reconciler.IPsLister.List, impl.EnqueueKey, &caCerts)
+	if err != nil {
+		logger.Fatal("Failed to created prober", zap.Error(err))
+	}
```

Comment:
Currently, TLS is not supported on namespaced broker and we haven't defined how (and if) we want to implement TLS for it (there are a few reasons like data plane can't mount secrets cross namespace, etc)

I'd not do this here since we need to discuss those issues/problems for namespaced broker